vowpalwabbit/options.h

```diff
@@ -119,6 +120,7 @@ struct base_option
   size_t m_type_hash;
   std::string m_help = "";
   std::string m_short_name = "";
+  std::string m_one_of_str = "";

```

Comment:
Why not store the set instead of serializing and deserializing later?

Reply:
The set can either be `int` or `string` which would require templating.

Reply:
Typed option is templated based on the value of the option, that seems like the logical place for this to live

Reply:
Changed to dynamic cast base option to typed option